From 418599cbe1bf4fde308a2f97391d5a5050effaee Mon Sep 17 00:00:00 2001
From: Lars Kanis <lars@greiz-reinsdorf.de>
Date: Mon, 6 May 2019 21:46:21 +0200
Subject: [PATCH 1/1] Fix build on 32 bit mingw with gcc-7.4.0

Compilation failed on this assertion:

In file included from parse.y:26:0:
internal.h:175:36: error: static assertion failed: "len: HERETERM_LENGTH_BITS == 22"
 # define STATIC_ASSERT(name, expr) _Static_assert(expr, #name ": " #expr)

It appears that gcc doesn't pack the bit fields fully, so that the
struct size is 20 instead of 16 expected bytes in size.
Decreasing the "length" bits by one convinces gcc to pack the struct to
16 bytes.
---
 parse.y | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parse.y b/parse.y
index fd0810987f..c7448c1b2d 100644
--- a/parse.y
+++ b/parse.y
@@ -608,7 +608,7 @@ typedef struct rb_strterm_literal_struct {
     } u3;
 } rb_strterm_literal_t;
 
-#define HERETERM_LENGTH_BITS ((SIZEOF_VALUE - 1) * CHAR_BIT - 1)
+#define HERETERM_LENGTH_BITS ((SIZEOF_VALUE - 1) * CHAR_BIT - 2)
 
 struct rb_strterm_heredoc_struct {
     VALUE lastline;	/* the string of line that contains `<<"END"` */
-- 
2.12.2.windows.2

